src/controllers/events.js

```diff
@@ -29,3 +30,16 @@ exports.getRelatedEvents = async (req, res, next) => {
     .populate('ngo', 'name');
   return res.status(200).json({ success: true, data: relatedEvents });
 };
+
+exports.addEvent = async (req, res, next) => {
+  // Create new event
+  const newEvent = await Event.create(req.body);
+  newEvent.ngo = req.user._id;
+  await newEvent.save();
```

Comment:
You could skip that extra `save` operation and instead define the `ngo` before creating the `event`.

```javascript
 req.body.ngo = req.user._id;
 const newEvent = await Event.create(req.body); 
 await newEvent.save(); // This line could be removed now

```